src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,10 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
+        {
+            return base.SingleTypeNameNotFound($"{whereText}' or '{whereText}Attribute", diagnostics, location, forwardedToAssembly);
```

Comment:
We can still do this with one error, by making the "or" a part of the message in .RESX file. A new error should be added in this case.